### PR TITLE
Add tests for sankey and minor fixes

### PR DESCRIPTION
**Pull request by oscargus**
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/sankey.py [hunk 1]
```diff
@@ -525,7 +521,7 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
             if orient == 1:
                 if is_input:
                     angles[i] = DOWN
-                elif not is_input:
+                elif is_input is False:
```
Comment (tacaswell) [inline lib/matplotlib/sankey.py:524]:
```suggestion
                elif not is_input:
```

File: lib/matplotlib/sankey.py [hunk 2]
```diff
@@ -538,7 +534,7 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
                         f"but it must be -1, 0, or 1")
                 if is_input:
                     angles[i] = UP
-                elif not is_input:
+                elif is_input is False:
```
Comment (tacaswell) [inline lib/matplotlib/sankey.py:537]:
```suggestion
                elif not is_input:
```

**Comment by tacaswell** [comment]
I think `not var` is both clearer and more idiomatic than `var is False` (I started to put in suggested edits on them, but gave up after 2).  `not var`  handles anything that will cast though bool (rather than being exactly the `False` object).

File: lib/matplotlib/sankey.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/sankey.py:524]: If I remember correctly, the original code actually breaks for None (I did this a few months ago, but pushed to the main repo #22916 ). See the comment on the next line.

Recall that `elif not is_input:` is simply `else:` as `not None` is `True`.

**Comment by tacaswell** [comment]
Never mind, I take back my comment.  These are internal values (not user input) and the values are a tri-state between {True, False, None}.  This has been broken from a4f1588462fb48aa13f79ae797e37155539240b5 .

**Comment by tacaswell** [comment]
Re-ordering all of the tests to check for `is None` first or switching to an Enum would be "better", but I do not think it is worth the effort.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
I have a minor preference for the commit to include a note about the non-idiomatic code.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**